File: packages/mobile/src/services/ai.ts
import { api } from './api';
import type { AISuggestPriorityResponse, SimilarReportItem } from '../types';

export async function suggestPriority(
  title: string,
  description: string
): Promise<AISuggestPriorityResponse> {
  const { data } = await api.post<AISuggestPriorityResponse>('/ai/suggest-priority', {
    title,
    description,
  });
  return data;
}

export async function enhanceDescription(
  title: string,
  description: string
): Promise<string> {
  const { data } = await api.post<{ enhanced_description: string }>('/ai/enhance-description', {
    title,
    description,
  });
  return data.enhanced_description;
}

export async function getSimilarReports(reportId: string): Promise<SimilarReportItem[]> {
  const { data } = await api.get<{ items: SimilarReportItem[] }>(`/reports/${reportId}/similar`);
  return data.items;
}

export async function suggestNote(reportId: string): Promise<string> {
  const { data } = await api.post<{ suggestion: string }>(`/reports/${reportId}/suggest-note`);
  return data.suggestion;
}

File: packages/backend/app/api/ai.py
import asyncio
from typing import Annotated

from fastapi import APIRouter, Depends

from app.api.deps import get_current_user
from app.models.user import User
from app.schemas.report import (
    AIEnhanceRequest,
    AIEnhanceResponse,
    AISuggestPriorityRequest,
    AISuggestPriorityResponse,
)
from app.services import ai_service

router = APIRouter(prefix="/ai", tags=["ai"])


@router.post("/suggest-priority", response_model=AISuggestPriorityResponse)
async def suggest_priority(
    data: AISuggestPriorityRequest,
    _current_user: Annotated[User, Depends(get_current_user)],
) -> AISuggestPriorityResponse:
    result = await asyncio.to_thread(ai_service.suggest_priority, data.title, data.description)
    return AISuggestPriorityResponse(**result)  # type: ignore[arg-type]


@router.post("/enhance-description", response_model=AIEnhanceResponse)
async def enhance_description(
    data: AIEnhanceRequest,
    _current_user: Annotated[User, Depends(get_current_user)],
) -> AIEnhanceResponse:
    enhanced = await asyncio.to_thread(ai_service.enhance_description, data.title, data.description)
    return AIEnhanceResponse(enhanced_description=enhanced)
